src/internal/SignalGenerator.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#ifndef SIGNAL_GENERATOR_H
+#define SIGNAL_GENERATOR_H
+
+namespace hyro
+{ 
+  enum Function {cosine, sine};
+  class SignalGenerator
```

Comment:
Usually we do not indent the class with respect to the namespace.
Check `include/hyro/DigitalConverterComponent.h`